R/set.R

```diff
@@ -0,0 +1,119 @@
+#' Set operations
+#'
+#' @description
+#' - `vec_set_intersect()` computes a set intersection. It returns the first
+#'   occurrence of values in `x` that also appear in `y`.
+#'
+#' - `vec_set_difference()` computes an asymmetric set difference. If returns
```

Comment:
Do you think `vec_set_symmetric_difference()` would be useful? I can add it in another PR, and we did add `dplyr::symdiff()` in dplyr 1.1.0 so it could use it too. I also have `ivs::iv_symmetric_difference()`

Reply:
I don't really mind either way as you can always do the union minus the intersection. I suppose a dedicated function could be faster. I don't love the length of that function name but I'm not sure how to do better.

It's a bit strange that a dplyr function does not use separators in its name by the way. Why not `sym_diff()`? What will the funs versions be called? I assume the dplyr function will be reexported from funs?

Reply:
I assume it is because we have `dplyr::setdiff()` and `dplyr::setequal()`, so we added `dplyr::symdiff()` with no `_`

These only work with data frames btw

Reply:
So we'll have `dplyr::symdiff()` that only works with data frames and `funs::?` that works with both data frames and vectors?

Reply:
Oh I mispoke, there is a `dplyr:::symdiff.default()` that uses the base functions

```r
> dplyr:::symdiff.default
function (x, y, ...) 
{
    check_dots_empty()
    setdiff(union(x, y), intersect(x, y))
}
```

And then there is a `dplyr:::symdiff.data.frame()` method which is supposedly more efficient than all that

```r
> dplyr:::symdiff.data.frame
function (x, y, ...) 
{
    check_dots_empty()
    check_compatible(x, y)
    cast <- vec_cast_common(x = x, y = y)
    only_x <- vec_slice(cast$x, !vec_in(cast$x, cast$y))
    only_y <- vec_slice(cast$y, !vec_in(cast$y, cast$x))
    out <- vec_unique(vec_rbind(only_x, only_y))
    dplyr_reconstruct(out, x)
}
```

Reply:
So that still leaves us in a place where `symdiff()` doesn't work with things like `Date`s, which seems undesirable for a dplyr function. I'm not quite sure how to handle this but I do think we should try to resolve it before 1.1.0 is released.